tapfreighter/interface.go

```diff
@@ -218,6 +221,35 @@ type Anchor struct {
 	NumPassiveAssets uint32
 }
 
+// OutputIdentifier is a key that can be used to uniquely identify a transfer
+// output.
+type OutputIdentifier [32]byte
+
+// NewOutputIdentifier creates a new output identifier from the proof suffix and
+// script key.
+func NewOutputIdentifier(proofSuffix []byte,
+	scriptKey btcec.PublicKey) (OutputIdentifier, error) {
+
+	var zero [32]byte
+	if len(proofSuffix) == 0 {
+		return zero, fmt.Errorf("proof suffix not set")
+	}
+
+	var outProofAsset asset.Asset
+	err := proof.SparseDecode(
+		bytes.NewReader(proofSuffix),
+		proof.AssetLeafRecord(&outProofAsset),
+	)
+	if err != nil {
+		return zero, fmt.Errorf("unable to sparse decode proof: %w",
+			err)
+	}
+
+	keyData := append([]byte{}, fn.ByteSlice(outProofAsset.ID())...)
+	keyData = append(keyData, scriptKey.SerializeCompressed()...)
```

Comment:
No outpoint needed?

Reply:
Hmm, I guess you're right, we should probably also add the output index (the TXID is superfluous because a transfer always represents one on-chain TX), since you could theoretically have the same asset ID with the same script key in multiple outputs (although I think that's what we currently disallow to fix the exclusion proof collision issue?). But adding the output index definitely shouldn't hurt, so going to add that.